localgov_eu_cookie_compliance.module

```diff
@@ -217,3 +248,146 @@ function _localgov_eu_cookie_compliance_defer_tracker_scripts(array &$attachment
 
   $attachment_list['#attached']['drupalSettings']['deferred_scripts'] = json_encode($deferred_head_items_markup);
 }
+
+/**
+ * Based on the implementation at https://git.drupalcode.org/project/cookies/-/blob/2.x/modules/cookies_video/cookies_video.module under GPLv2.
+ *
+ * Implements hook_preprocess_HOOK().
+ */
+function localgov_eu_cookie_compliance_preprocess_field(&$variables) {
+  $formatter = $variables["element"]["#formatter"];
+  if ($formatter === 'oembed') {
+    localgov_eu_cookie_compliance_media_oembed_handler($variables);
+  }
+}
+
+/**
+ * Handling oembed field formatters.
+ *
+ * @param array $variables
+ *   Field template variables (s. cookies_video_preprocess_field()).
+ */
+function localgov_eu_cookie_compliance_media_oembed_handler(array &$variables) {
+  $formatter = $variables["element"]["#formatter"];
+  /** @var \Drupal\Core\Field\FieldItemList $field_item */
+  $field_item = $variables['element']['#items'];
+  /** @var \Drupal\media\Entity\Media $media */
+  $media = $variables['element']['#object'];
+  $media_id = $media->id();
+  $variables['attributes']['class'][] = "media-oembed-$media_id";
+  $field_value = $field_item->getValue();
+  if (isset($field_value) && !empty($field_value)) {
+    $field_value = reset($field_value);
+    $video_src = $field_value['value'];
+  }
+
+  // Get the oembed settings lines and split them out.
+  $oembed_settings = Drupal::config('localgov_eu_cookie_compliance.settings')->get('disabled_oembeds') ?? "";
+  $disabled_oembeds = explode("\n", trim($oembed_settings));
+
+  // Convert the domains with optional category into an associative
+  // array of domain to category.
+  $blocked_domains = [];
+  foreach ($disabled_oembeds as $embed) {
+    // Extract the category condition, if present.
+    $parts = explode(':', $embed);
+    if (count($parts) == 1) {
+      $condition = "";
+      $embed = $parts[0];
+    }
+    else {
+      $condition = $parts[0];
+      $embed = end($parts);
+    }
+    $blocked_domains[$embed] = $condition;
+  }
+
+  foreach ($variables["items"] as &$item) {
+    switch ($formatter) {
+      case 'oembed':
+        // Get the video src being embedded.
+        $item["content"]["#attributes"]['data-src'] = $video_src;
```

Comment:
I have a feeling this will overwrite all video URLs in a multi-value field with the very first video.  This is based on what I see on line 281 above.

Reply:
Just so I'm sure I understand, do you mean when there's a media field that's set to allow more than one entry, and multiple remote videos are added, or something else?

Reply:
> when there's a media field that's set to allow more than one entry, and multiple remote videos are added

That's right.